Sources/Fuzzilli/Util/WeakSet.swift

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+struct WeakVar<T: AnyObject>: Hashable where T: Hashable {
+    weak var value: T?
+
+    static func == (_ lhs: WeakVar, _ rhs: WeakVar) -> Bool {
+        return lhs.value == rhs.value
+    }
+
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(value)
+    }
+}
+
+public struct WeakSet<Element: AnyObject> where Element: Hashable {
+    private var elements: Set<WeakVar<Element> > = []
+
+    public var count: Int {
```

Comment:
Could it make sense to not even expose a `count` for this since it might be ambiguous whether the count includes nil values or excludes them? The count only seems to be used in tests currently.

Reply:
yeah I agree, but I still would like to test `removeNils`, so maybe I can expose `nilCount` which returns number of nil elements?

Reply:
I see. Yeah that also sounds like a good idea. I think you can also expose it just as `internal` protection level (just remove `public`, `internal` is the default) if its (for now) only used in tests